core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -901,7 +901,7 @@ class DAGScheduler(
       // the stage as completed here in case there are no tasks to run
       markStageAsFinished(stage, None)
 
-      val debugString = stage match {
+      def debugString: String = stage match {
```

Comment:
I don't think so, because `logDebug` is declared as `logDebug(msg: => String)`; the laziness is built in. As in...


```
scala> def foo(eval: Boolean, msg: => String) = if (eval) println(msg)
foo: (eval: Boolean, msg: => String)Unit

scala> val i = new java.util.concurrent.atomic.AtomicInteger()
i: java.util.concurrent.atomic.AtomicInteger = 0

scala> foo(true, s"${i.incrementAndGet}")
1

scala> i
res5: java.util.concurrent.atomic.AtomicInteger = 1

scala> foo(false, s"${i.incrementAndGet}")

scala> i
res7: java.util.concurrent.atomic.AtomicInteger = 1
```

It wasn't evaluated in the second instance.